File: runtime/lib/platform.ts
import type { OsOptions } from '@torba/core';

export function currentPlatform(
  platform: string = process.platform,
  arch: string = process.arch,
): OsOptions {
  return {
    name:
      platform === 'win32'
        ? 'windows'
        : platform === 'darwin'
          ? 'osx'
          : 'linux',
    version: '',
    arch: arch === 'arm64' ? 'aarch64' : 'x86_64',
  };
}

File: crates/opys-runtime/src/platform.rs
use opys_core::OsOptions;

pub fn current_platform() -> OsOptions {
    let name = if cfg!(target_os = "windows") {
        "windows"
    } else if cfg!(target_os = "macos") {
        "osx"
    } else {
        "linux"
    };
    let arch = if cfg!(target_arch = "aarch64") {
        "aarch64"
    } else {
        "x86_64"
    };
    OsOptions {
        name: name.into(),
        version: String::new(),
        arch: arch.into(),
    }
}
